extensions/skins/conversation_v1/Conversation.js

```diff
@@ -18,41 +18,184 @@
+    if (className !== 'animate-card-change') {
+      return;
+    }
+
+    var currentHeight = element.height();
+    var expectedNextHeight = $('#futurePreview').height();
+
+    // Fix the current card height, so that it does not change during the
+    // animation, even though its contents might.
+    element.css('height', currentHeight);
+
+    jQuery(element).animate({
+      opacity: 0
+    }, TIME_FADEOUT_MSEC).animate({
+      height: expectedNextHeight
+    }, TIME_HEIGHT_CHANGE_MSEC).animate({
+      opacity: 1
+    }, TIME_FADEIN_MSEC, function() {
+      element.css('height', '');
+      done();
+    });
+
+    return function(cancel) {
+      if (cancel) {
+        element.css('opacity', '1.0');
+        element.css('height', '');
+        element.stop();
+      }
+    };
+  };
+
+  return {
+    addClass: animateCardChange
+  };
+});
+
 // TODO(sll): delete/deprecate 'reset exploration' from the list of
 // events sent to a container page.
-
 oppia.directive('conversationSkin', [function() {
   return {
     restrict: 'E',
     scope: {},
     templateUrl: 'skins/Conversation',
     controller: [
-        '$scope', '$timeout', '$rootScope', '$window', '$modal', 'warningsData',
-        'messengerService', 'oppiaPlayerService', 'urlService', 'focusService',
-        'ratingService',
+        '$scope', '$timeout', '$rootScope', '$window', 'messengerService',
+        'oppiaPlayerService', 'urlService', 'focusService', 'ratingService',
+        'windowDimensionsService',
         function(
-          $scope, $timeout, $rootScope, $window, $modal, warningsData,
-          messengerService, oppiaPlayerService, urlService, focusService,
-          ratingService) {
+          $scope, $timeout, $rootScope, $window, messengerService,
+          oppiaPlayerService, urlService, focusService, ratingService,
+          windowDimensionsService) {
+
+      // The minimum width, in pixels, needed to be able to show two cards
+      // side-by-side.
+      $scope.TWO_CARD_THRESHOLD_PX = 1120;
+      $scope.CONTINUE_BUTTON_FOCUS_LABEL = 'continueButton';
+
+      // The minimum number of milliseconds that should elapse before Oppia
+      // responds, so that the transition isn't too sudden.
+      var MIN_WAIT_TIME_MILLISECS = 1000;
 
       var hasInteractedAtLeastOnce = false;
-      var _labelForNextFocusTarget = null;
       var _answerIsBeingProcessed = false;
-      var _learnerInputIsInView = false;
+      var _nextFocusLabel = null;
 
       $scope.isInPreviewMode = oppiaPlayerService.isInPreviewMode();
-      $scope.introCardImageUrl = null;
-
-      $rootScope.loadingMessage = 'Loading';
       $scope.isIframed = urlService.isIframed();
+      $rootScope.loadingMessage = 'Loading';
+      // This will be replaced with the dataURI representation of the
+      // user-uploaded profile image, if it exists.
+      $scope.profilePicture = '/images/general/user_blue_72px.png';
+      $scope.finished = false;
```

Comment:
'finished' is a vague variable name. Please rename it to include what is finished.